lib/core/network/executor.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:chopper/chopper.dart' show Response;
+import 'package:coffeecard/core/errors/failures.dart';
+import 'package:dartz/dartz.dart';
+import 'package:logger/logger.dart';
+
+class Executor {
+  final Logger logger;
+
+  const Executor(this.logger);
+
+  /// Executes a network request.
+  Future<Either<ServerFailure, Result>> call<Result>(
```

Comment:
i think this adds excessive function calls that are not needed, and bloats any calling function. The class already contains only a single function that only does one thing. Otherwise the class could be renamed to NetworkExecutor or RequestExecutor.